lib/session.rb

```diff
@@ -0,0 +1,30 @@
+# typed: true
+# frozen_string_literal: true
+
+module AutoHCK
+  class Session < T::Struct
+    extend T::Sig
+    extend Models::JsonHelper
+
+    prop :common, CommonOptions
+    prop :test, TestOptions
+    prop :install, InstallOptions
+    prop :mode, T.nilable(String), default: nil
```

Comment:
At https://github.com/HCK-CI/AutoHCK/pull/487#discussion_r1981046821, I suggested adding a property for `CLI` instead of properties that `CLI` has.